semver.py

```diff
@@ -548,17 +548,19 @@ def __getitem__(self, index):
 
         if (
             isinstance(index, slice)
-            and (index.start is None or index.start < 0)
-            and (index.stop is None or index.stop < 0)
+            and (index.start is not None and index.start < 0)
+            or (index.stop is not None and index.stop < 0)
         ):
             raise IndexError("Version index cannot be negative")
 
-        # Could raise IndexError:
-        part = tuple(filter(None, self.to_tuple()[index]))
+        part = tuple(filter(lambda p: p is not None, self.to_tuple()[index]))
 
         if len(part) == 1:
             part = part[0]
-        if not part:
+        elif len(part) == 0:
+            raise IndexError("Version part undefined")
+
+        if part is None:
             raise IndexError("Version part undefined")
```

Comment:
I don't like that we need to raise IndexError two times. Could we combine it? Like this:

```suggestion
        elif part is None or len(part) == 0:
            raise IndexError("Version part undefined")
```

It works for me, can you confirm?

Reply:
That does work fine. I'm going to apply it.